### PR TITLE
Document the concept of 'coarse cell id' in the glossary.

### PR DESCRIPTION
**Pull request by bangerth**
@peterrum -- FYI. Do you know of other places I should add a link to the glossary entry to?

### CODE REVIEW

**Comment by peterrum** [comment]
Looks very well! Thanks for the work!

File: doc/doxygen/headers/glossary.h [hunk 1]
```diff
@@ -413,6 +413,67 @@
...
  *   they are active (i.e., have no children) or have been refined is not
  *   important for this definition.)
+ *
+ *   Most of the triangulation classes in deal.II store the entire coarse
+ *   mesh along with at least some of the refined cells. (Both the
+ *   dealii::Triangulation and parallel::shared::Triangulation classes
+ *   actually store <i>all</i> cells of the entire mesh, whereas some
+ *   other classes such as parallel::distributed::Triangulation only
+ *   store <i>some</i> of the @ref GlossActive "active cells" on
+ *   each process in a parallel computation.) In those, cases,
```
Comment (kronbichler) [inline doc/doxygen/headers/glossary.h]:
```suggestion
 *   each process in a parallel computation.) In those cases,
```

File: doc/doxygen/headers/glossary.h [hunk 2]
```diff
@@ -413,6 +413,67 @@
...
+ *   @ref GlossCoarseCellId "the concept of coarse cell ids"
+ *   for that case.
+ * </dd>
+ *
+ *
+ * <dt class="glossary">@anchor GlossCoarseCellId <b>Coarse cell ID</b></dt>
+ * <dd>
+ *   Most of the triangulation classes in deal.II, notably
+ *   dealii::Triangulation, parallel::shared::Triangulation, and
+ *   parallel::distributed::Triangulation store the entire
```
Comment (kronbichler) [inline doc/doxygen/headers/glossary.h]:
```suggestion
 *   parallel::distributed::Triangulation, store the entire
```

**Comment by bangerth** [comment]
Thanks, @kronbichler. I've updated the patch with your suggestions.

**Pull request merged**